grades/api_test.py

```diff
@@ -213,6 +236,29 @@ def test_compute_grade_for_non_fa_certs(self, certificate_type, status, grade_re
         assert grade.grade == self.certificates.get(course_key).data.get('grade')
         assert grade.payed_on_edx is payed_edx_result
 
+    @ddt.data('', None, ' ', 'foo', )
+    def test_compute_grade_for_non_fa_odd_grade(self, odd_value):
+        """
+        Tests for _compute_grade_for_non_fa function in case the grade coming from edX is not a number
+        """
+        course_key1 = self.run_no_fa.edx_course_key
+        current_grade1 = self.current_grades[course_key1]
+        current_grade1.data['percent'] = odd_value
+        current_grade1.save()
+        user_edx_data1 = CachedEdxUserData(self.user)
+        run_data1 = user_edx_data1.get_run_data(course_key1)
+        grade1 = api._compute_grade_for_non_fa(run_data1)
+        assert grade1.grade == 0.0
+
+        course_key2 = self.run_no_fa_with_cert.edx_course_key
+        cert2 = self.certificates[course_key2]
+        cert2.data['grade'] = odd_value
+        cert2.save()
+        user_edx_data2 = CachedEdxUserData(self.user)
+        run_data2 = user_edx_data2.get_run_data(course_key2)
+        grade2 = api._compute_grade_for_non_fa(run_data2)
+        assert grade2.grade == 0.0
```

Comment:
it would be 2 different tests, but each test would be shorter by about half, and we wouldn't be repeating the same logic. if you prefer, it would also be possible to limit it to one test case and loop over tuples:

```python
for course_run, course_run_with_cert in [(self.run_no_fa, self.run_no_fa_with_cert), (self.run_fa, self.run_fa_with_cert)]:
```